src/main/scala/com/exasol/cloudetl/kafka/consumer/KafkaRecordConsumer.scala

```diff
@@ -0,0 +1,119 @@
+package com.exasol.cloudetl.kafka.consumer
+
+import java.time.Duration
+import java.util.Arrays
+
+import scala.jdk.CollectionConverters._
+
+import com.exasol.ExaIterator
+import com.exasol.ExaMetadata
+import com.exasol.cloudetl.kafka._
+import com.exasol.cloudetl.kafka.deserialization._
+
+import com.typesafe.scalalogging.LazyLogging
+import org.apache.kafka.clients.consumer.ConsumerRecords
+import org.apache.kafka.clients.consumer.KafkaConsumer
+import org.apache.kafka.common.TopicPartition
+
+/**
+ * A class that polls data from Kafka topic and emits records into an
+ * Exasol table.
+ */
+final case class KafkaRecordConsumer(
+  properties: KafkaConsumerProperties,
+  partitionId: Int,
+  partitionStartOffset: Long,
+  tableColumnCount: Int,
+  nodeId: Long,
+  vmId: String
+) extends LazyLogging {
+
+  private[this] val topic = properties.getTopic()
+  private[this] val consumer = getRecordConsumer()
+  private[this] val partitionEndOffset = getPartitionEndOffset()
```

Comment:
I would also inject it as a field

Reply:
Same here, since consumer is not outside, it can also stay.